src/base/utils/fs.cpp

```diff
@@ -72,9 +72,26 @@ QString Utils::Fs::toUniformPath(const QString &path)
     return QDir::fromNativeSeparators(path);
 }
 
-/**
- * Returns the file extension part of a file name.
- */
+QString Utils::Fs::normalizeDirPath(const QString &path)
```

Comment:
IMO path (in filesystem sense) should only have one form: don't end with a trailing slash.
If this can be uphold through out the code base, then concatenating file path and file name is easy, just add slash (appropriate path separator) and append the file name and we don't need this helper.

Reply:
This logic is just moved from another place. I don't want to change it in this PR.
Currently, I have unfinished changes that encapsulate the path handling in a separate class, which will save us from unnecessary cares throughout the code base. But this job requires a lot of routine code changes (due to the extensive use of paths in qBittorrent), so I decided to finish it after my currently opened PRs are done.

Reply:
Anyway trying to throw it away here...